common/rpc/interceptor/dc_redirection_policy.go

```diff
@@ -188,8 +188,7 @@ func (policy *SelectedAPIsForwardingRedirectionPolicy) getTargetClusterAndIsName
 	}
 
 	if len(namespaceEntry.ClusterNames()) == 1 {
-		// do not do dc redirection if namespace is only targeting at 1 dc (effectively local namespace)
-		return policy.currentClusterName, false
+		return namespaceEntry.ActiveClusterName(), true
```

Comment:
we can just remove this if?

Reply:
The following code can still return current non-active cluster if there is only 1 cluster in namespaceEntry.

Reply:
I think if the auto forwarding is not enabled, then we should still use current cluster?

Reply:
removed per offline discussion